speech/recognize.js

```diff
@@ -23,143 +23,275 @@
 
 'use strict';
 
-const Speech = require('@google-cloud/speech');
+function syncRecognize (filename, encoding, sampleRate) {
+  // [START speech_sync_recognize]
+  // Imports the Google Cloud client library
+  const Speech = require('@google-cloud/speech');
 
-// [START speech_sync_recognize]
-function syncRecognize (filename) {
   // Instantiates a client
   const speech = Speech();
 
-  const config = {
-    // Configure these settings based on the audio you're transcribing
-    encoding: 'LINEAR16',
-    sampleRate: 16000
+  // The path to the local file on which to perform speech recognition, e.g. /path/to/audio.raw
+  // const filename = '/path/to/audio.raw';
+
+  // The encoding of the audio file, e.g. 'LINEAR16'
+  // const encoding = 'LINEAR16';
+
+  // The sample rate of the audio file, e.g. 16000
+  // const sampleRate = 16000;
+
+  const request = {
+    encoding: encoding,
+    sampleRate: sampleRate
   };
 
-  // Detects speech in the audio file, e.g. "./resources/audio.raw"
-  return speech.recognize(filename, config)
+  // Detects speech in the audio file
+  speech.recognize(filename, request)
     .then((results) => {
       const transcription = results[0];
+
+      console.log(`Transcription: ${transcription}`);
+    });
+  // [END speech_sync_recognize]
+}
+
```

Comment:
FYI. Missing JSDoc for the functions - could be nice to have `@param` blocks with suggested values, e.g. '16000' for samplerate, 'LINEAR16' for encoding. Feel free to ignore.